e2e/framework/machine.go

```diff
@@ -54,3 +60,46 @@ func FilterRunningMachines(machines []*clusterv1.Machine) []*clusterv1.Machine {
 
 	return result
 }
+
+// GetMachine get a machine by its name from the cluster API namespace.
+func GetMachine(cl client.Client, name string) (*clusterv1.Machine, error) {
+	machine := &clusterv1.Machine{}
+	key := client.ObjectKey{Namespace: CAPINamespace, Name: name}
+
+	if err := cl.Get(ctx, key, machine); err != nil {
+		return nil, fmt.Errorf("error querying api for machine object: %w", err)
+	}
+
+	return machine, nil
+}
+
+
+// DeleteMachines deletes the specified machines and returns an error on failure.
+func DeleteMachines(cl client.Client, machines ...*clusterv1.Machine) error {
```

Comment:
What about using the gomega primitives here? I think it's more readable + will keep our e2es looking like our units / integrations (less mental load to debug / review)

```go
for _, machine := range machines {
  Eventually(func() error {
    return cl.Delete(ctx, machine)
  }, time.Minute, RetryShort).Should(SatisfyAny(
    Succeed(),
    WithTransform(apierrors.IsNotFound, BeTrue()),
  ), "Delete should succeed, or machine should not be found")
}
```

Reply:
This `Eventually` also removes the need for the wait func below (no longer needed)